backend/clickhouse/logs.go

```diff
@@ -0,0 +1,112 @@
+package clickhouse
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	e "github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type LogRow struct {
+	Timestamp       time.Time
+	SeverityText    string
+	Body            string
+	ProjectId       uint32
+	SecureSessionID string
+}
+
+// func (client *Client) CreateLogsTable(ctx context.Context) error {
+// 	client.conn.Exec(ctx, "DROP TABLE IF EXISTS logs") //nolint:errcheck
+
+// 	return client.conn.Exec(ctx, `
+// 	CREATE TABLE IF NOT EXISTS logs (
+// 		Timestamp       DateTime64(9) CODEC (Delta, ZSTD(1)),
+// 		SeverityText    LowCardinality(String) CODEC (ZSTD(1)),
+// 		Body            String CODEC (ZSTD(1)),
+// 		ProjectId 		UInt32 CODEC (ZSTD(1)),
+// 		SecureSessionID Nullable(String) CODEC (ZSTD(1))
+// 	)
+// 	ENGINE = MergeTree()
+// 		PARTITION BY toDate(Timestamp)
+// 		ORDER BY (ProjectId, toUnixTimestamp(Timestamp))
+// 		TTL toDateTime(Timestamp) + toIntervalDay(30)
+// 		SETTINGS index_granularity = 8192, ttl_only_drop_parts = 1;
+// 	`)
+// }
+
+type Message struct {
+	Type  string `json:"type"`
+	Trace []struct {
+		ColumnNumber int    `json:"columnNumber"`
+		LineNumber   int    `json:"lineNumber"`
+		FileName     string `json:"fileName"`
+		FunctionName string `json:"functionName,omitempty"`
+		Source       string `json:"source"`
+	} `json:"trace"`
+	Value []string `json:"value"`
+	Time  int64    `json:"time"`
+}
+
+type Messages struct {
+	Messages []Message `json:"messages"`
+}
+
+func (client *Client) BatchWriteMessagesForSession(ctx context.Context, projectID int, sessionSecureID string, messages string) error {
```

Comment:
nit: i would prefer to have a separate function here for parsing / unmarshalling the `messages` string and doing the clickhouse batched write itself. that way, if we have a structured (typed) source of the messages, that can use the `BatchWrite` here and reuse this code path.

maybe there is a `BatchWriteMessages` that takes `[]Messages` and `BatchWriteMessagesForSession` calls `BatchWriteMessages` after unmarshalling?

Reply:
I see what you're saying. I'm just not sure when we'd be using `[]Message` outside the context of a session.
Let's see what this looks like when we add additional logging ingestions and we can tweak the design as needed.